src/vs/base/common/errors.ts

```diff
@@ -239,11 +239,18 @@ export class ExpectedError extends Error {
  * Error that when thrown won't be logged in telemetry as an unhandled error.
  */
 export class ErrorNoTelemetry extends Error {
-	override readonly name: string;
+	private _name = 'ErrorNoTelemetry';
 
 	constructor(msg?: string) {
 		super(msg);
-		this.name = 'ErrorNoTelemetry';
+	}
+
+	public override set name(newName: string) {
+		this._name = `ErrorNoTelemetry${newName}`;
+	}
+
+	public override get name(): string {
```

Comment:
My worry with this change is that we might have existing code that probes on `error.name` to figure out the file service error:

<img width="290" alt="image" src="https://user-images.githubusercontent.com/900690/180364639-824b3c1d-d33e-4131-9e53-22ac74743211.png">

(above is a "Find Referenfes" on `error.name`).

For example here:

https://github.com/microsoft/vscode/blob/13ba7bb446a638d37ebccb1a7d74e31c32bb9790/src/vs/workbench/api/common/extHostFileSystemConsumer.ts#L149-L164

and

https://github.com/microsoft/vscode/blob/13ba7bb446a638d37ebccb1a7d74e31c32bb9790/src/vs/workbench/api/browser/mainThreadFileSystem.ts#L140-L162

In other words, this may change how extensions can detect which error was thrown?

Reply:
We tried that but found when errors got serialized across processes that it would be only the base error properties that were left and therefore we had to use one of the three error properties and figured name was the least important one. @sbatten and I have tried many solutions but our architecture is quite complex so ends up breaking the ErrorNoTelemetry one way or another. It seems the name probing is unique to FileSystemErrors

Reply:
Yeah, so file system errors flow from processes into the ext host but also from extensions back into the renderer.